src/components/Popover/index.tsx

```diff
@@ -0,0 +1,265 @@
+import clsx from 'clsx';
+import { makeStyles, Theme } from '@material-ui/core';
+import { Popover as MaterialPopover, PopoverProps as MaterialPopoverProps } from '@material-ui/core';
+import IconButton from '@material-ui/core/IconButton';
+import Close from '@material-ui/icons/Close';
+import { useUniqueId } from '../../hooks/useUniqueId';
+
+const useStyles = makeStyles((theme: Theme) => ({
+    body: {
+        padding: '0.8rem 1rem',
+    },
+    '.body-size-small': {
+        maxWidth: '210px',
+        '&.fixed-width': {
+            width: '210px',
+        },
+    },
+    '.body-size-medium': {
+        maxWidth: '310px',
+        '&.fixed-width': {
+            width: '310px',
+        },
+    },
+    '.body-size-large': {
+        maxWidth: '460px',
+        '&.fixed-width': {
+            width: '460px',
+        },
+    },
+    container: {
+        zIndex: 2000,
+    },
+    content: {
+        contentOverflowVisible: true,
+        flex: '1 1 auto',
+        color: theme.palette.grey['700'],
+    },
+    dismiss: {
+        flex: '0 0 auto',
+        fontSize: '20px',
+        order: 1,
+        padding: '0',
+    },
+    hasDismiss: {
+        display: 'flex',
+        alignItems: 'flex-start',
+    },
+    header: {
+        flex: '1 1 auto',
+        '& h2': {
+            margin: 0,
+            paddingBottom: '0.5em',
+        },
+    },
+    trigger: {
+        display: 'inline-block',
+        color: 'inherit',
+    },
+    'trigger-type-text': {
+        border: 0,
+        margin: 0,
+        padding: 0,
+        backgroundColor: 'transparent',
+        cursor: 'pointer',
+        borderBottom: '1px dashed currentColor',
+        '&:focus': {
+            outline: 'none',
+        },
+    },
+}));
+
+export interface PopoverProps {
+    /**
+     * Determines where the popover is displayed when opened, relative to the trigger.
+     * If the popover doesn't have enough space to open in this direction, it
+     * automatically chooses a better direction based on available space.
+     */
+    position?: 'top' | 'right' | 'bottom' | 'left';
+
+    /**
+     * Determines the maximum width for the popover.
+     */
+    size?: 'small' | 'medium' | 'large';
+
+    /**
+     * Expands the popover body to its maximum width regardless of content.
+     * For example, use it when you need to place a column layout in the popover content.
+     */
+    fixedWidth?: boolean;
+
+    /**
+     * Specifies the type of content inside the trigger region. The following types are available:
+     * - `text` - Use for inline text triggers.
+     * - `custom` - Use for the Northstar Button component.
+     */
+    triggerType?: 'text' | 'custom';
+
+    /**
+     * Element that triggers the popover when selected by the user.
+     * @displayname trigger
+     */
+    children?: React.ReactNode;
+
+    /**
+     * Specifies optional header text for the popover.
+     */
+    header?: string;
```

Comment:
Maybe this should be a ReactNode so users can pass in their `<Heading>` etc to choose how it's shown?